lib/ui/pages/news_list/home_page.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:allthenews/ui/common/util/dimens.dart';
+import 'package:allthenews/ui/common/util/strings.dart';
+import 'package:allthenews/ui/common/widget/primary_text_button.dart';
+import 'package:allthenews/ui/pages/news_list/primary_news/primary_news_list_entity.dart';
+import 'package:allthenews/ui/pages/news_list/primary_news/primary_news_list_view.dart';
+import 'package:allthenews/ui/pages/news_list/secondary_news/secondary_news_list_entity.dart';
+import 'package:allthenews/ui/pages/news_list/news_list_page.dart';
+import 'package:allthenews/ui/pages/news_list/secondary_news/secondary_news_list_view.dart';
+import 'package:flutter/material.dart';
+
+abstract class _Constants {
+  static const sectionHeaderPadding = 10.0;
+}
+
+class HomePage extends StatefulWidget {
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Container(
+              width: double.maxFinite,
+              height: 80,
+            ),
+            _buildNewsSectionHeader(
+              title: Strings.mostViewed,
+              routeBuilder: (context) => NewsListPage(
+                headerTitle: Strings.mostViewed,
+                listEntities: primaryNewsListEntities
```

Comment:
jeżeli SecondaryNewsListEntity jest używany zarówno w kontekście primaryNews jak i secondaryNews to wydaje mi się, że powinien mieć jakąś inną nazwę, bo nieco dziwnie wygląda jak primaryNews mapowany jest na SecondaryNewsViewEntity żeby wysłać do NewsListPage listę SecondaryNewsListEntity - skoro NewsListPage (nieważne czy primary czy secondary) to może NewsViewEntity?

Reply:
to też jest moja bolączka z tym nazewnictwem. Myślałem o NewsViewEntity, ale wtedy dlaczego elementy na horyzontalnej liście miałyby się tak nie nazywać? Trochę zaczyna mi nie pasować też samo "primary" i "secondary", bo miało to sens jak mieliśmy dwie listy na "home" screenie, a teraz jak jedna z tych list przechodzi w inną to już się to miesza

Reply:
Rozróżnianie ich domenowo też się nie sprawdza, bo po przejściu do "See all" będziemy wrzucać do listy różne domenowo itemy. A co powiesz na nazewnictwo typu Large i Small ? Można zrobić, że : 
- duże klocki będą reprezentowały encję UserFeedNews, a element wyświetlany na liście horyzontalnej to byłby LargeNewsListItem,
- małe klocki będą reprezentowane przez encję LatestNews, a element wyświetlany na liście to będzie SmallNewsListItem,
- wtedy będzie trzeba stworzyć osobne klasy, które będą wrzucane do listy, żeby UserFeedNews był mapowany do SmalNewsListItem, a nie do LatestNews.

Odpowiedź na to pytanie w sumie wiąże się z innymi pytaniami w tym PR dot struktury plików w projekcie

Co myślisz?

Reply:
Odnośnie rozróżniania domenowego to właśnie chodziło mi o to, żeby ta lista do której wrzucamy dwa rodzaje newsów była niezależna od domeny :P ale faktycznie samo newsListViewEntity brzmi średnio z uwagi na istnienie tego dużego elementu. Co do propozycji spoko, pasuje, aczkolwiek jeżeli primary i secondary nie dotyczą domeny, bo domeną będzie UserFeedNews i LatestNews to teoretycznie tego problemu już nie ma, ale faktycznie zastanawiam się czy to ja trochę się za mocno focusowałem, że primary secondary to domena czy one po prostu faktycznie brzmią tak że je można traktować domenowo? Też tak trochę do tego podchodziłeś, czy nazwy primary/secondary traktowałeś czysto ui? - tak z ciekawości pytam :P Co do propozycji to jestem za, large/small na pewno nie pomylą się z domeną :D

Reply:
Traktowałem primary i secondary UIowo, patrzyłem na to w kontekście tego że domena dopiero dojdzie, tym bardziej że same stringi są w tych klasach. Ok to zrobię tak jak pisałem :)

Reply:
bardziej chodziło mi o to, że myślałem, że domena to będzie coś typu PrimaryNews i SecondaryNews bo te viewEntity to wiadomo, że nie domena ;)

Reply:
myślę, że userfeed i latest więcej mówią o tym co tam rzeczywiście jest w tych obiektach. Primary i secondary to tak jak jeden i dwa

Reply:
ciągle się zastanawiam nad tym small/large. SmallNewsListItem i SmallNewsListViewEntity - o ile ListItem ok, to jak pomyślałem o SmallNewsViewEntity to średnio - może jednak primary i secondary? xD tak głośno myślę :D

Reply:
hmm nie no dobrze, trzeba to przegadać bo ewidentnie jest niejasne. Z tym small/large to miałem w głowie, że już nie będzie pośrednich encji tylko sam listItem będzie miał przekazywane to co ma wyświetlić jako pola, ale chyba rzeczywiście lepiej cały obiekt mu dawać.

Nie rozumiem dlaczego potrzebujemy dwa obiekty `SmallNewsListViewEntity` i `SmallNewsViewEntity`? 

Teraz mamy `SecondaryNewsListEntity` i `PrimaryNewsListEntity` - klasy trzymające dane do wrzucenia na widok. Są one przekazywane odpowiednio do `SecondaryNewsListItem` i `PrimaryNewsListItem` - klasy zawierające sam widok.

Załóżmy, że zostajemy przy primary i secondary. Wtedy mapowanie z primary na secondary w celu wrzucenia do listy nie wydaje się takie złe i wracamy do punktu wyjścia. Możemy jedynie zrobić dodatkowo klasy domenowe `UserFeedNews` i `LatestNews` i je mapować odpowiednio do `PrimaryNewsListEntity` i/lub `SecondaryNewsListEntity`.

Reply:
wolę to przegadać zanim zacznę to implementować :)

Reply:
Mamy PrimaryListView, który zawiera PrimaryListView, który zawiera PrimaryListItem - jakby zmiana na SmallNewsListItem i LargeNewsListItem pociągnie za sobą ListView i ViewEntity, czy się mylę, dlatego też zacząłem zastanawiać się czy to że jest obecnie tak że primaryNewsViewEntity mapuje się na secondaryNewsViewEntity to coś złego, na początku źle traktowałem to nazewnictwo, bo słowo primary traktowałem jako coś co jest oddomenowe czyli że primaryViewEntity to jest encja dla widoku dla głównych newsów, że domena to `główne newsy` - ale jak już moja wizja została zrewidowana to może lepiej zostawić tak jak jest? najwyżej jak któremuś wpadnie kiedyś jakiś super pomsył to nic nie stoi na przeszkodzie aby zmienić nazwenictwo, co ty na to? :)

Reply:
trochę chaotycznie napisałeś, zwłaszcza z tymi nazwami ,ale chyba kumam o ocb, ok to zostańmy przy Primary secondary zwlaszcza ze domenowych modeli jeszcze nie ma